addons/misra.py

```diff
@@ -468,7 +468,11 @@ def getEssentialTypeCategory(expr):
         return "enum<" + expr.valueType.typeScope.className + ">"
     vartok = expr
     while simpleMatch(vartok, '[') or (vartok and vartok.str == '*' and vartok.astOperand2 is None):
-        vartok = vartok.astOperand1
+        if simpleMatch(vartok.astOperand1, '.'):
```

Comment:
It could handle multiple dots like so:

    while simpleMatch(vartok, '[') or (vartok and vartok.str == '*' and vartok.astOperand2 is None):
        vartok = vartok.astOperand1
        while simpleMatch(vartok, '.'):
            vartok = vartok.astOperand2

Reply:
Good point! Will update + add multisite test

Reply:
while ((vartok && simpleMatch(vartok, '[')) || (vartok && vartok->str == '*' && vartok->astOperand2 == nullptr)) {
    vartok = vartok->astOperand1;
    while (vartok && simpleMatch(vartok, '.')) {
        vartok = vartok->astOperand2;
    }
}